pkg/storage/registry/spaces/spaces.go

```diff
@@ -348,15 +344,21 @@ func (r *registry) findProvidersForFilter(ctx context.Context, filters []*provid
 					appctx.GetLogger(ctx).Error().Err(err).Interface("provider", provider).Interface("space", space).Msg("failed to execute template, continuing")
 					continue
 				}
+				if match, err := regexp.MatchString(sc.MountPoint, spacePath); err != nil || !match {
+					continue
+				}
+
 				setPath(space, spacePath)
 				validSpaces = append(validSpaces, space)
 			}
 
-			if err := setSpaces(p, validSpaces); err != nil {
-				appctx.GetLogger(ctx).Debug().Err(err).Interface("provider", provider).Interface("spaces", validSpaces).Msg("marshaling spaces failed, continuing")
-				continue
+			if len(validSpaces) > 0 {
+				if err := setSpaces(p, validSpaces); err != nil {
+					appctx.GetLogger(ctx).Debug().Err(err).Interface("provider", provider).Interface("spaces", validSpaces).Msg("marshaling spaces failed, continuing")
+					continue
+				}
+				providerInfos = append(providerInfos, p)
 			}
-			providerInfos = append(providerInfos, p)
```

Comment:
There is a code path that tries to fetch only provdier info. Let me see if I can dig that up ...

Reply:
Doesn't matter. Mount points should always be obeyed.

Assume I have two providers mounted at `/users/[a-k]` and `/users/[l-z]`. When `alice` wants to list personal spaces, she should just get one space located on the first provider. At the moment, she also gets `/users/Alice` mounted on the second one

Can you please look into making the tests pass?